src/main/java/ch/njol/skript/lang/Expression.java

```diff
@@ -239,6 +239,20 @@ default boolean canReturn(Class<?> returnType) {
 	 */
 	int getTime();
 
+	/**
+	 * Allows nested structures for this expression, i.e. lists of lists.
+	 * <p>
+	 * <b>Note</b>:
+	 * Nested structures must be flattened in {@link #getArray(Event)} and {@link #getAll(Event)},
+	 * i.e. if this expression returns a list of lists of players,
+	 * {@link #getArray(Event)} must return a single array containing all players of all lists
+	 *
+	 * @return Whether this expression allows nested structures.
+	 */
+	default boolean allowNestedStructures() {
```

Comment:
Should it be possible to disallow nested structures? This method changing state on the Expression is slightly odd to me, but not necessarily a major issue.